privacyidea/lib/resolvers/LDAPIdResolver.py

```diff
@@ -638,6 +651,64 @@ def testconnection(cls, param):
         
         return success, desc
 
+    def _attributes_to_ldap_attributes(self, attributes, uid):
+        """
+        takes the attributes and maps them to the LDAP attributes
+        :param attributes: Attributes to be updated
+        :type attributes: dict
+        :param uid: The uid of the user object in the resolver
+        :type uid: basestring
+        :return: dict with attribute name as keys and values
+        """
+        ldap_attributes = {}
+        for fieldname, value in attributes.iteritems():
+            if self.map.get(fieldname):
+                if fieldname == "password":
+                    password = value 
+                    # Create a {SSHA} password
+                    salt = geturandom(4)
+                    hr = hashlib.sha1(password)
+                    hr.update(salt)
+                    ldap_attributes[self.map.get(fieldname)] = \
+                        [MODIFY_REPLACE, ["{SSHA}" + hr.digest() + salt]]
+                else:
+                    if value:
+                        if fieldname in self.getUserInfo(uid):
+                            ldap_attributes[self.map.get(fieldname)] = \
+                                 [MODIFY_REPLACE, [value]]
+                        else:
+                            ldap_attributes[self.map.get(fieldname)] = \
+                                 [MODIFY_ADD, [value]]
+                    else:
+                        ldap_attributes[self.map.get(fieldname)] = \
+                             [MODIFY_DELETE, [value]]
+
+        return ldap_attributes
+
+    def update_user(self, uid, attributes=None):
+        """
+        Update an existing user.
+        This function is also used to update the password. Since the
+        attribute mapping know, which field contains the password,
+        this function can also take care for password changing.
+
+        Attributes that are not contained in the dict attributes are not
+        modified.
+
+        :param uid: The uid of the user object in the resolver.
+        :type uid: basestring
+        :param attributes: Attributes to be updated.
+        :type attributes: dict
+        :return: True in case of success
+        """
+        attributes = attributes or {}
+        self._bind()
+
+        params = self._attributes_to_ldap_attributes(attributes, uid)
+        self.l.modify(uid, params)
```

Comment:
I've been thinking about this over night and I don't think it will be too difficult. Here are my thoughts:
1. Have l.modify/add/delete methods write a additional results dictionary to the existing tmp_directory file tests/testdata/tmp_directory
2. Have the result method read this information from that file
3. l.search I don't believe uses the result method

Best Regards

Martin